### PR TITLE
fix(bpm-bench): concurrency issues

### LINKED ISSUE #518: Concurrency issues in `bpm-bench`
# `FetchPage`

It is assumed that `FetchPage` does not fail and cannot return `nullptr` (e.g., when the BP is busy):

https://github.com/cmu-db/bustub/blob/e0c109516cfe2003e3c5c7db2f000eccde23dc5f/tools/bpm_bench/bpm_bench.cpp#L170-L172

https://github.com/cmu-db/bustub/blob/e0c109516cfe2003e3c5c7db2f000eccde23dc5f/tools/bpm_bench/bpm_bench.cpp#L199-L201

If you change these parameters to `2` and `1`, respectively, you can observe a segmentation fault:
https://github.com/cmu-db/bustub/blob/e0c109516cfe2003e3c5c7db2f000eccde23dc5f/tools/bpm_bench/bpm_bench.cpp#L35-L36

# Page read/writes
Page read/writes need to be synchronised by latches here:
https://github.com/cmu-db/bustub/blob/e0c109516cfe2003e3c5c7db2f000eccde23dc5f/tools/bpm_bench/bpm_bench.cpp#L173-L176

https://github.com/cmu-db/bustub/blob/e0c109516cfe2003e3c5c7db2f000eccde23dc5f/tools/bpm_bench/bpm_bench.cpp#L202

### PR DESCRIPTION
**Pull request by CuriousGeorgiy**
Closes #518

### CODE REVIEW

File: tools/bpm_bench/bpm_bench.cpp [hunk 1]
```diff
@@ -197,14 +202,18 @@ auto main(int argc, char **argv) -> int {
...
 
+        page->RLatch();
         char ch = page->GetData()[page_idx % 1024];
+        page->RUnlatch();
         if (ch == 0) {
           throw std::runtime_error("invalid data");
         }
 
         bpm->UnpinPage(page->GetPageId(), false, AccessType::Get);
-        page_idx += 1;
```
Comment (skyzh) [inline tools/bpm_bench/bpm_bench.cpp:207]: we will need to keep this line in order to iterate all pages?

**Review by skyzh — COMMENTED** [review]
rest LGTM, thanks!

File: tools/bpm_bench/bpm_bench.cpp (hunk 1 above)
Comment (CuriousGeorgiy) [reply tools/bpm_bench/bpm_bench.cpp:207]: No, as I mentioned in the commit message, the addition result is never used, as the variable is initialized at the beginning of the loop:
https://github.com/cmu-db/bustub/blob/71a0168dc490b3e361c592328843c64e74453819/tools/bpm_bench/bpm_bench.cpp#L198

**Review by skyzh — APPROVED** [review]

**Comment by skyzh** [comment]
Thanks!

**Pull request merged**